addons/sale/models/sale.py

```diff
@@ -520,7 +520,7 @@ def action_confirm(self):
             order.message_subscribe([order.partner_id.id])
         self.write({
             'state': 'sale',
-            'confirmation_date': fields.Datetime.now()
+            'confirmation_date': self.confirmation_date or fields.Datetime.now(),
```

Comment:
This change is wrong as the method is decorated with ``@api.multi``

Reply:
Furthermore why would we write the confirmation date as it's supposed to be unset at the confirmation ?